searx/webapp.py

```diff
@@ -1067,8 +1068,7 @@ def image_proxy():
     if not url:
         return '', 400
 
-    h = new_hmac(settings['server']['secret_key'], url.encode())
-    if h != request.args.get('h'):
+    if not is_hmac_of(settings['server']['secret_key'], url.encode(), request.args.get('h', '')):
```

Comment:
Bug fix: `request.args.get('h', '')` instead of `request.args.get('h')` so the value is always a string.